### PR TITLE
refactor(map_projection_loader): rework parameters

### PR DESCRIPTION
**Pull request by TaikiYamada4**
# **This PR must be merged with https://github.com/autowarefoundation/autoware_launch/pull/839**
## Description

This PR has done three changes.
1. Extract ros parameters from `map_projection_loader.launch.xml` to `map_projection_loader.param.yaml`.
2. Create `map_projection_loader.schema.json` to validate `map_projection_loader.param.yaml`.
3. Let `README.md` to reflect ros parameters from `map_projection_loader.schema.json`.

### 1. Passing ros parameters

I did the [rosbag replay simulation tutorial](https://autowarefoundation.github.io/autoware-documentation/main/tutorials/ad-hoc-simulation/rosbag-replay-simulation/) and checked.
```
ros2 param dump /map/map_projection_loader
```
and got the follwoing result which is expected.
```
/map/map_projection_loader:
  ros__parameters:
    lanelet2_map_path: /home/taikiyamada/autoware_map/sample-map-rosbag/lanelet2_map.osm
    map_projector_info_path: /home/taikiyamada/autoware_map/sample-map-rosbag/map_projector_info.yaml
    qos_overrides:
      /clock:
        subscription:
          depth: 1
          durability: volatile
          history: keep_last
          reliability: best_effort
      /parameter_events:
        publisher:
          depth: 1000
          durability: volatile
          history: keep_last
          reliability: reliable
    use_sim_time: true

```

### 2. JSON validation
I did the following command at `autoware.universe/map/` and confirmed that `map_projection_loader.schema.json` and `map_projection_loader.param.yaml` are OK.
```
find -wholename '*/schema/*.schema.json' -printf '%p: ' -execdir bash -c 'check-jsonschema --schemafile "$1" ../config/"${1:2:-12}"*.param.yaml' bash '{}' +
```

### 3. Write a table of ros parameters via `map_projection_loader.schema.json`
I did `mkdocs serve` and confirmed that the table of ros parameters is constructed in the documents.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: map/map_projection_loader/README.md [hunk 1]
```diff
@@ -82,11 +82,10 @@ map_origin:
...
-- ~/map_projector_info (tier4_map_msgs/MapProjectorInfo) : Topic for defining map projector information
+- `~/map_projector_info` (tier4_map_msgs/MapProjectorInfo) : This topic shows the definition of map projector information
 
 ## Parameters
 
-| Name                    | Type        | Description                                                                      |
-| :---------------------- | :---------- | :------------------------------------------------------------------------------- |
-| map_projector_info_path | std::string | A path to map_projector_info.yaml (used by default)                              |
-| lanelet2_map_path       | std::string | A path to lanelet2 map (used only when `map_projector_info_path` does not exist) |
+Note that these paramters are assumed to be passed from launch arguments, and it is not recommended to directly write them in `map_projection_loader.param.yaml`.
```
Comment (SakodaShintaro) [inline map/map_projection_loader/README.md:89]: There seems to be a typographical error.

**Review by SakodaShintaro — COMMENTED** [review]
I have completed checking the schema and confirming the operation in my environment. I will approve if you correct the typo.

File: map/map_projection_loader/README.md (hunk 1 above)
Comment (TaikiYamada4) [reply map/map_projection_loader/README.md]: Thanks for letting me know!
I fixed it now.

**Review by SakodaShintaro — APPROVED** [review]

**Comment by HansRobo** [comment]
I updated this branch to import #6278.
This will help you pass `build-and-test-differential` faster.

**Pull request merged**